skins/laika/src/components/pages/donation_form/Address.vue

```diff
@@ -81,143 +77,31 @@ export default Vue.extend( {
-				},
-				country: {
-					name: 'country',
-					value: 'DE',
-					pattern: this.$props.addressValidationPatterns.country,
-					optionalField: false,
-				},
-				email: {
-					name: 'email',
-					value: '',
-					pattern: this.$props.addressValidationPatterns.email,
-					optionalField: false,
-				},
-			},
-		};
-	},
 	props: {
 		validateAddressUrl: String,
 		validateEmailUrl: String,
 		validateBankDataUrl: String,
 		validateLegacyBankDataUrl: String,
-		countries: Array as () => Array<Country>,
+		countries: Array as PropType<Array<Country>>,
 		isDirectDebit: Boolean,
-		addressValidationPatterns: Object as () => AddressValidation,
-	},
-	computed: {
-		fieldErrors: {
-			get: function (): AddressValidity {
-				return Object.keys( this.formData ).reduce( ( validity: AddressValidity, fieldName: string ) => {
-					if ( !this.formData[ fieldName ].optionalField ) {
-						validity[ fieldName ] = this.$store.state.address.validity[ fieldName ] === Validity.INVALID;
-					}
-					return validity;
-				}, ( {} as AddressValidity ) );
-			},
-		},
-		disabledAddressTypes: {
-			get: function (): Array<AddressTypeModel> {
-				return this.$store.getters[ 'payment/isDirectDebitPayment' ] ? [ AddressTypeModel.ANON ] : [];
-			},
-		},
-		...mapGetters( NS_ADDRESS, [
-			'addressType',
-			'addressTypeIsNotAnon',
-			'addressTypeIsInvalid',
-		] ),
-		addressTypeName(): string {
-			return addressTypeName( this.$store.state.address.addressType );
-		},
-		receiptNeeded(): Boolean {
-			return !this.$store.state.address.receiptOptOut;
-		},
-	},
-	mounted() {
-		Object.entries( this.$data.formData ).forEach( ( formItem ) => {
-			const key: string = formItem[ 0 ];
-			this.$data.formData[ key ].value = this.$store.state.address.values[ key ];
-			if ( this.$store.state[ NS_ADDRESS ].validity[ key ] === Validity.RESTORED ) {
-				this.$store.dispatch( action( NS_ADDRESS, validateAddressField ), this.$data.formData[ key ] );
-			}
-		} );
+		addressValidationPatterns: Object as PropType<AddressValidation>,
 	},
-	methods: {
-		validateForm(): Promise<ValidationResult> {
-			return Promise.all( [
-				this.$store.dispatch( action( NS_ADDRESS, validateAddressType ), this.$store.state.address.addressType ),
-				this.$store.dispatch( action( NS_ADDRESS, validateAddress ), this.$props.validateAddressUrl ),
-				this.$store.dispatch( action( NS_ADDRESS, validateEmail ), this.$props.validateEmailUrl ),
-			] ).then( mergeValidationResults );
-		},
-		onFieldChange( fieldName: string ): void {
-			this.$store.dispatch( action( NS_ADDRESS, setAddressField ), this.$data.formData[ fieldName ] );
-		},
-		onAutofill( autofilledFields: { [key: string]: string; } ) {
-			Object.keys( autofilledFields ).forEach( key => {
-				const fieldName = camelizeName( key );
-				if ( this.$data.formData[ fieldName ] ) {
-					this.$store.dispatch( action( NS_ADDRESS, setAddressField ), this.$data.formData[ fieldName ] );
+	// TODO properly type 'props'
```

Comment:
Typed props would be really great here (and in `useAddressFunctions`). My first attempt to solve this would be to use `defineComponent` from the composition API + a props interface instead of using `Vue.extend`. See https://stackoverflow.com/questions/59966387/strongly-typing-props-of-vue-components-using-composition-api-and-typescript-typ

The store can stay untyped for now, that's not additional tech debt but one whe had for some time.